app/src/main/java/com/yacgroup/yacguide/AscendActivity.kt

```diff
@@ -45,63 +47,60 @@ import java.util.Calendar
 class AscendActivity : AppCompatActivity() {
 
     private lateinit var _db: DatabaseWrapper
-    private var _ascend: Ascend? = null
+    private lateinit var _ascend: Ascend
+    private var _existingAscend: Ascend? = null
     private var _route: Route? = null
-    private var _partnerIds: ArrayList<Int>? = null
-    private var _styleId: Int = 0
-    private var _year: Int = 0
-    private var _month: Int = 0
-    private var _day: Int = 0
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_ascend)
 
         _db = DatabaseWrapper(this)
 
-        _ascend = _db.getAscend(intent.getIntExtra(IntentConstants.ASCEND_KEY, DatabaseWrapper.INVALID_ID))
+        // The database query is null, if this activity is called from the route DescriptionActivity.
+        _existingAscend = _db.getAscend(intent.getIntExtra(IntentConstants.ASCEND_KEY,
```

Comment:
It's quite confusing having 2 `Ascend` objects here. Wouldn't it work to just set `_ascend = _db.getAscend(...from_ASCEND_KEY...) ?: Ascend()`? The field `id` is initialized by `0` if we create a new ascend according to file `Ascend.kt`. So instead of checking for `_existingAscend` in method `enter()` you could check for `_ascend.id == 0`. Doing so, we could get rid of this confusing `_existingAscend` variable.